src/entities/notesubfolder.cpp

```diff
@@ -303,13 +304,24 @@ QVector<NoteSubFolder> NoteSubFolder::fetchAllByParentId(
  * @return
  */
 QVector<int> NoteSubFolder::fetchIdsRecursivelyByParentId(int parentId) {
+    const QSqlDatabase db = QSqlDatabase::database(QStringLiteral("memory"));
+    QSqlQuery query(db);
+
     QVector<int> idList = QVector<int>{parentId};
-    const auto noteSubFolders = fetchAllByParentId(parentId);
-    idList.reserve(noteSubFolders.count());
 
-    for (const NoteSubFolder& noteSubFolder : noteSubFolders) {
-        const int id = noteSubFolder.getId();
-        idList << fetchIdsRecursivelyByParentId(id);
+    const QString& sql = QStringLiteral(
+        "SELECT id FROM noteSubFolder WHERE parent_id = "
```

Comment:
Are you sure this does the same as `fetchIdsRecursivelyByParentId` on all of `fetchAllByParentId`? Your query doesn't get the ids recursively... It basically only substitutes `fetchAllByParentId` alone.

Reply:
Yes, i will fix it.